xpallets/assets-registrar/src/lib.rs

```diff
@@ -19,118 +19,59 @@ pub mod weights;
 use sp_std::prelude::*;
 
 use frame_support::{
-    decl_error, decl_event, decl_module, decl_storage,
     dispatch::{DispatchError, DispatchResult},
     ensure,
     log::info,
-    traits::Get,
-    IterableStorageMap,
 };
-use frame_system::ensure_root;
 
 use chainx_primitives::{AssetId, Desc, Token};
 
+#[cfg(feature = "std")]
+use frame_support::traits::GenesisBuild;
+
 pub use self::types::AssetInfo;
 pub use self::weights::WeightInfo;
 pub use xp_assets_registrar::{Chain, RegistrarHandler};
 
-/// The module's config trait.
-///
-/// `frame_system::Config` should always be included in our implied traits.
-pub trait Config: frame_system::Config {
-    /// The overarching event type.
-    type Event: From<Event> + Into<<Self as frame_system::Config>::Event>;
-
-    /// Native asset Id.
-    type NativeAssetId: Get<AssetId>;
+pub use pallet::*;
 
-    /// Handler for doing stuff after the asset is registered/deregistered.
-    type RegistrarHandler: RegistrarHandler;
-
-    /// Weight information for extrinsics in this pallet.
-    type WeightInfo: WeightInfo;
-}
-
-decl_event!(
-    /// Event for the XAssetRegistrar Module
-    pub enum Event {
-        /// A new asset was registered. [asset_id, has_mining_rights]
-        Registered(AssetId, bool),
-        /// A deregistered asset was recovered. [asset_id, has_mining_rights]
-        Recovered(AssetId, bool),
-        /// An asset was deregistered. [asset_id]
-        Deregistered(AssetId),
-    }
-);
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    use frame_support::pallet_prelude::*;
+    use frame_system::pallet_prelude::*;
 
-decl_error! {
-    /// Error for the XAssetRegistrar Module
-    pub enum Error for Module<T: Config> {
-        /// Token symbol length is zero or too long
-        InvalidAssetTokenSymbolLength,
-        /// Token symbol char is invalid, only allow ASCII alphanumeric character or '-', '.', '|', '~'
-        InvalidAssetTokenSymbolChar,
-        /// Token name length is zero or too long
-        InvalidAssetTokenNameLength,
-        /// Desc length is zero or too long
-        InvalidAssetDescLength,
-        /// Text is invalid ASCII, only allow ASCII visible character [0x20, 0x7E]
-        InvalidAscii,
-        /// The asset already exists.
-        AssetAlreadyExists,
-        /// The asset does not exist.
-        AssetDoesNotExist,
-        /// The asset is already valid (online), no need to recover.
-        AssetAlreadyValid,
-        /// The asset is invalid (not online).
-        AssetIsInvalid,
-    }
-}
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(_);
```

Comment:
`pub struct Pallet<T>(_);` won't make your compilation fail, but it's better to use `pub struct Pallet<T>(PhantomData<T>);` instead. `PhantomData` is there to let you tell the compiler information that it cannot infer itself, correctly expressing lifetimes in `PhantomData` is a part of creating a safe abstraction around unsafe code.